janus_server/src/datastore.rs

```diff
@@ -909,11 +909,104 @@ impl Transaction<'_> {
         )
     }
 
+    /// Returns the task ID for the provided collect job ID, or `None` if no such collect job
+    /// exists.
+    #[tracing::instrument(skip(self), err)]
+    pub(crate) async fn get_collect_job_task_id(
+        &self,
+        collect_job_id: Uuid,
+    ) -> Result<Option<TaskId>, Error> {
+        let stmt = self
+            .tx
+            .prepare_cached(
+                "SELECT task_id FROM tasks
+                WHERE id = (SELECT task_id FROM collect_jobs WHERE collect_job_id = $1)",
+            )
+            .await?;
+        self.tx
+            .query_opt(&stmt, &[&collect_job_id])
+            .await?
+            .map(|row| TaskId::get_decoded(row.get("task_id")).map_err(Error::from))
+            .transpose()
+    }
+
+    /// Returns the collect job for the provided UUID, or `None` if no such collect job exists.
+    #[tracing::instrument(skip(self), err)]
+    pub(crate) async fn get_collect_job<A: vdaf::Aggregator, E>(
+        &self,
+        collect_job_id: Uuid,
+    ) -> Result<Option<CollectJob<A>>, Error>
+    where
+        A: vdaf::Aggregator,
+        E: std::fmt::Display,
+        for<'a> A::AggregateShare: TryFrom<&'a [u8], Error = E>,
+        for<'a> &'a A::AggregateShare: Into<Vec<u8>>,
+    {
+        let stmt = self
+            .tx
+            .prepare_cached(
+                "SELECT
+                    task_id, batch_interval_start, batch_interval_duration, aggregation_param,
+                    helper_aggregate_share, leader_aggregate_share, report_count, checksum
+                FROM collect_jobs
+                WHERE collect_job_id = $1;",
+            )
+            .await?;
+        let collect_job_row = match self.tx.query_opt(&stmt, &[&collect_job_id]).await? {
+            Some(row) => row,
+            None => return Ok(None),
+        };
+        let task_id: i64 = collect_job_row.get("task_id");
+
+        let stmt = self
+            .tx
+            .prepare_cached("SELECT task_id FROM tasks WHERE id = $1")
+            .await?;
```

Comment:
To avoid the need for a second network round-trip, retrieve tasks.task_id with the original query via a join or subquery. Check out several of the other `get_foo` for example implementations.